app/controllers/v0/sessions_controller.rb

```diff
@@ -97,6 +95,17 @@ def saml_callback
 
     private
 
+    def stats(status = :ok)
```

Comment:
I like where you're going with moving the stats, but I don't think it fixes the problem that caused me to revert in #2730. From skimming the code in statsd-instrument, my theory is that I we always need to pas the same number of tags for the metric, and now It's passing 2 tags for failure for and 3 for success STATSD_SSO_CALLBACK_KEY.  Also, I'm suspecting we'd want to update the initializer in config/initializers/statsd.rb.  The catch is that it all works in dev/test because we're using `StatsD::Instrument::Backends::LoggerBackend` rather than a real statsd implementation.

Reply:
i made some more changes, should always be 4 values now. I think i already made the changes in initializer but you can verify.

Reply:
I made some changes that should address these points, including changes to the initializer. Could you verify?